lib/adapters/QUnitAdapter.js

```diff
@@ -1,22 +1,52 @@
 import EventEmitter from "../EventEmitter.js";
 import {Test, Suite} from "../Data.js";
 
-/*global QUnit*/
 export default class QUnitAdapter extends EventEmitter {
-    constructor() {
+    constructor(QUnit) {
         super();
-        QUnit.done(this.onDone.bind(this));
+        this.QUnit = QUnit;
+        QUnit.begin(this.onBegin.bind(this));
+        QUnit.moduleStart(this.onModuleStart.bind(this));
+        QUnit.testStart(this.onTestStart.bind(this));
+        QUnit.log(this.onLog.bind(this));
         QUnit.testDone(this.onTestDone.bind(this));
         QUnit.moduleDone(this.onModuleDone.bind(this));
-        QUnit.log(this.onLog.bind(this));
-        QUnit.testStart(this.onTestStart.bind(this));
+        QUnit.done(this.onDone.bind(this));
+
+        this.idDataMap = {};
+    }
+
+    convertModule(qunitModule) {
+        var suiteName = qunitModule.name;
+        var tests = [];
+        for (let qunitTest of qunitModule.tests) {
+            let test = new Test(qunitTest.name, suiteName);
+            tests.push(test);
+            this.idDataMap[qunitTest.testId] = test;
+        }
+
+        var suite = new Suite(suiteName, [], tests);
+        this.idDataMap[qunitModule.name] = suite;
+        return suite;
+    }
 
-        this.tests = {};
-        this.suites = [];
+    onBegin() {
+        var suites = [];
+        for (let suite of this.QUnit.config.modules) {
```

Comment:
Let's add a comment here, something like "// Access QUnit internals to build liste of modules, working around missing event data"